src/Consolonia.GuiCS/WindowsDriver.cs

```diff
@@ -2,294 +2,63 @@
-			try {
-				ReadConsoleInput (InputHandle, pRecord, bufferSize,
-					out var numberEventsRead);
-
-				return numberEventsRead == 0
-					? null
-					: new [] { Marshal.PtrToStructure<InputRecord> (pRecord) };
-			} catch (Exception) {
-				return null;
-			} finally {
-				Marshal.FreeHGlobal (pRecord);
-			}
-		}
-
-		[StructLayout (LayoutKind.Sequential)]
-		public struct CONSOLE_SCREEN_BUFFER_INFOEX {
-			public uint cbSize;
-			public Coord dwSize;
-			public Coord dwCursorPosition;
-			public ushort wAttributes;
-			public SmallRect srWindow;
-			public Coord dwMaximumWindowSize;
-			public ushort wPopupAttributes;
-			public bool bFullscreenSupported;
-
-			[MarshalAs (UnmanagedType.ByValArray, SizeConst = 16)]
-			public COLORREF [] ColorTable;
-		}
-
-		[StructLayout (LayoutKind.Explicit, Size = 4)]
-		public struct COLORREF {
-			public COLORREF (byte r, byte g, byte b)
-			{
-				Value = 0;
-				R = r;
-				G = g;
-				B = b;
-			}
-
-			public COLORREF (uint value)
-			{
-				R = 0;
-				G = 0;
-				B = 0;
-				Value = value & 0x00FFFFFF;
-			}
-
-			[FieldOffset (0)]
-			public byte R;
-			[FieldOffset (1)]
-			public byte G;
-			[FieldOffset (2)]
-			public byte B;
-
-			[FieldOffset (0)]
-			public uint Value;
-		}
+    [SupportedOSPlatform("windows")]
+    internal class WindowsConsole
+    {
+
+        internal HFILE InputHandle;
+        readonly CONSOLE_INPUT_MODE originalConsoleMode;
+
+        public WindowsConsole()
+        {
+            InputHandle = GetStdHandle(StdHandleType.STD_INPUT_HANDLE);
+            originalConsoleMode = ConsoleMode;
+            var newConsoleMode = originalConsoleMode;
+            newConsoleMode |= (CONSOLE_INPUT_MODE.ENABLE_MOUSE_INPUT |
+                                     CONSOLE_INPUT_MODE.ENABLE_EXTENDED_FLAGS);
+            newConsoleMode &= ~CONSOLE_INPUT_MODE.ENABLE_QUICK_EDIT_MODE;
+            newConsoleMode &= ~CONSOLE_INPUT_MODE.ENABLE_PROCESSED_INPUT;
+            ConsoleMode = newConsoleMode;
+        }
+
+        public CONSOLE_INPUT_MODE ConsoleMode
+        {
+            get
+            {
+                if (!GetConsoleMode(InputHandle, out CONSOLE_INPUT_MODE v))
+                    throw GetLastError().GetException();
+                return v;
+            }
+            set
+            {
+                if (!SetConsoleMode(InputHandle, value))
+                    throw GetLastError().GetException();
+            }
+        }
+
+        public INPUT_RECORD[] ReadConsoleInput()
+        {
+            const int bufferSize = 1;
+            var records = new INPUT_RECORD[bufferSize];
+            try
+            {
+                Kernel32.ReadConsoleInput(InputHandle, records, bufferSize,
+                    out var numberEventsRead);
+
+                return numberEventsRead == 0
+                    ? null
+                    : records;
+            }
+            catch (Exception)
```

Comment:
Exception should be specified. If it's Win32 or COM better to check code too

Reply:
hm. this was the original Gui.cs code, switching to  throw GetLastError().GetException();